library/core-utils/src/main/java/im/vector/lib/core/utils/timer/CountUpTimer.kt

```diff
@@ -55,17 +56,25 @@ class CountUpTimer(private val intervalInMs: Long = 1_000) {
     }
 
     fun pause() {
+        tickListener?.onTick(addAndGetElapsedTime())
         resumed.set(false)
     }
 
     fun resume() {
+        lastTime.set(clock.epochMillis())
         resumed.set(true)
     }
 
     fun stop() {
+        tickListener?.onTick(addAndGetElapsedTime())
         coroutineScope.cancel()
     }
 
+    private fun addAndGetElapsedTime(): Long {
+        val now = clock.epochMillis()
+        return elapsedTime.addAndGet(now - lastTime.getAndSet(now))
+    }
```

Comment:
Would it be more accurate to keep track of the start time and calculating each time:

```
elapsed time = now - start
```